cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -309,10 +309,24 @@ func (c *Client) GetClusterVersions() (map[string]string, error) {
 
 // ClusterAvailable returns true if the cluster status indicates the cluster is available.
 func (c *Client) ClusterAvailable() (bool, error) {
-	_, err := c.GetClusterStatus()
```

Comment:
but at this point this serves as code duplication.
i will defer to the opinion of @fabriziopandini here.

Reply:
I thought about adding a flag, noResponseMap, to Client#GetClusterStatus so that empty map is returned when noResponseMap is true.